server/proxy/install.go

```diff
@@ -157,6 +158,38 @@ func (p *Proxy) ensureOAuthApp(r *incoming.Request, conf config.Config, app *app
 	return nil
 }
 
+func (p *Proxy) createAndValidateBot(r *incoming.Request, bot *model.Bot) error {
+	mm := p.conf.MattermostAPI()
+
+	err := mm.Bot.Create(bot)
```

Comment:
I'm wondering if we could instead use `EnsureBot` with the profile image option supported by that API. I'm also wondering why that method never caused issues when used in `OnActivate` in plugins. This change LGTM either way